src/widgets/barchart.rs

```diff
@@ -1238,4 +1108,193 @@ mod tests {
+            ("i", 8),
+        ])
+            .into();
+        group = group.label(Line::from("Group").alignment(Alignment::Center));
+
+        let chart = BarChart::default()
+            .data(group)
+            .bar_set(symbols::bar::NINE_LEVELS);
+
+        let mut buffer = Buffer::empty(Rect::new(0, 0, 17, 3));
+        chart.render(buffer.area, &mut buffer);
+
+        assert_buffer_eq!(
+            buffer,
+            Buffer::with_lines(vec![
+                "  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8",
+                "a b c d e f g h i",
+                "      Group      ",
+            ])
+        );
+    }
+
+    #[test]
+    fn three_lines_double_width() {
+        let mut group = BarGroup::from(&[
+            ("a", 0),
+            ("b", 1),
+            ("c", 2),
+            ("d", 3),
+            ("e", 4),
+            ("f", 5),
+            ("g", 6),
+            ("h", 7),
+            ("i", 8),
+        ]);
+        group = group.label(Line::from("Group").alignment(Alignment::Center));
+
+        let chart = BarChart::default()
+            .data(group)
+            .bar_width(2)
+            .bar_set(symbols::bar::NINE_LEVELS);
+
+        let mut buffer = Buffer::empty(Rect::new(0, 0, 26, 3));
+        chart.render(buffer.area, &mut buffer);
+
+        assert_buffer_eq!(
+            buffer,
+            Buffer::with_lines(vec![
+                "   1▁ 2▂ 3▃ 4▄ 5▅ 6▆ 7▇ 8█",
+                "a  b  c  d  e  f  g  h  i ",
+                "          Group           ",
+            ])
+        );
+    }
+
+    #[test]
+    fn four_lines() {
+        let mut group: BarGroup = (&[
+            ("a", 0),
+            ("b", 1),
+            ("c", 2),
+            ("d", 3),
+            ("e", 4),
+            ("f", 5),
+            ("g", 6),
+            ("h", 7),
+            ("i", 8),
+        ])
+            .into();
+        group = group.label(Line::from("Group").alignment(Alignment::Center));
+
+        let chart = BarChart::default()
+            .data(group)
+            .bar_set(symbols::bar::NINE_LEVELS);
+
+        let mut buffer = Buffer::empty(Rect::new(0, 0, 17, 4));
+        chart.render(buffer.area, &mut buffer);
+
+        assert_buffer_eq!(
+            buffer,
+            Buffer::with_lines(vec![
+                "          ▂ ▄ ▆ █",
+                "  ▂ ▄ ▆ 4 5 6 7 8",
+                "a b c d e f g h i",
+                "      Group      ",
+            ])
+        );
+    }
+
+    #[test]
+    fn tow_lines_without_bar_labels() {
+        let group = BarGroup::default()
+            .label(Line::from("Group").alignment(Alignment::Center))
+            .bars(&[
+                Bar::default().value(0),
+                Bar::default().value(1),
+                Bar::default().value(2),
+                Bar::default().value(3),
+                Bar::default().value(4),
+                Bar::default().value(5),
+                Bar::default().value(6),
+                Bar::default().value(7),
+            ]);
```

Comment:
nit: let's go to 8 here for consistency? If we want to check how the rounding of values that don't exactly divide into the max, perhaps do so in a separate test method?

Reply:
> let's go to 8 here for consistency

done

> If we want to check how the rounding of values that don't exactly divide into the max, perhaps do so in a separate test method?

I am not sure if I understood you well, can you check whether the test function `one_lines_with_more_bars` does what asked?